app/routes/user/Tabs.tsx

```diff
@@ -0,0 +1,79 @@
+import type { FC } from 'react'
+
+type TabsProps = {
+  tabs: {
+    label: string
+    index: number
+    Component: FC<{ index: number; clientId: string; clientSecret: string }>
+    clientId: string
+    clientSecret: string
+  }[]
+  selectedTab: number
+  onClick: (index: number) => void
+  orientation?: 'horizontal' | 'vertical'
+  className?: string
+  clientId?: string
+  clientSecret?: string
+}
+
+/**
+ * Avalible Props
+ * @param className string
+ * @param tab Array of object
+ * @param selectedTab number
+ * @param onClick Function to set the active tab
+ * @param orientation Tab orientation Vertical | Horizontal
+ */
+let Tabs: FC<TabsProps> = ({
+  className = 'tabs-component',
+  tabs = [],
+  selectedTab = 0,
+  onClick,
+  orientation = 'horizontal',
+  clientId = '',
+  clientSecret = '',
```

Comment:
I think that it is likely that we will reuse this tabbed interface elsewhere on there site. Is it possible to refactor it so that the Tabs class does not need to know about the props of the child components like clientId and clientSecret?